File: 2020/5.php
<?php

require('libs/core.php');

$data = load_data("5.txt");

$seats = [];

foreach ($data as $key => $value) {
    $value = str_replace(["F", "B", "L", "R"], ["0", "1", "0", "1"], $value);
    $row = bindec(substr($value, 0, 7));
    $column = bindec(substr($value, 7));
    $id = $row * 8 + $column;
    $seats[] = $id;
}

sort($seats);

$max = end($seats);
$min = reset($seats);

print "Max is {$max}\n";

for ($i=$min; $i < $max; $i++) {
    if (!in_array($i, $seats)) {
       print "Seat $i is missing\n";
    }
}

File: 2020/python/5.py
import core
import re

data = core.load_data("5.txt")

seats = []

for value in data:
	value = re.sub("[FL]", "0", value)
	value = re.sub("[BR]", "1", value)
	row = int(value[0:7], 2)
	column = int(value[7:], 2)
	id = row * 8 + column
	seats.append(id)

seats.sort()

min = seats[0]
max = seats[-1]

print("Max is %d" % max)

for i in range(min, max):
	if not i in seats:
		print("Seat %d is missing" % i)
